parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -438,7 +438,15 @@ ensureMigPartition(){
 [Service]
 Environment="GPU_INSTANCE_PROFILE=${GPU_INSTANCE_PROFILE}"
 EOF
-    systemctlEnableAndStart mig-partition || exit $ERR_SYSTEMCTL_START_FAIL
+    # Noticed on Mariner that "nvidia-smi -mig 1" does not always enable MIG mode successfully and
+    # requires a reboot to fully actiavte MIG mode. Thus, as a workaround, we'll only run the MIG enable
+    # operation here and do the rest of the GPU instances creation after the reboot
+    if [[ $OS == $MARINER_OS_NAME ]]; then
+        retrycmd_if_failure 24 5 120 nvidia-smi -mig 1 || exit $ERR_GPU_DRIVERS_START_FAIL
```

Comment:
did we find out if ubuntu has the same issue?

Reply:
On Ubuntu "nvidia-smi -mig 1" also does not immediately enable MIG unless performing a reboot, which is the same as on Mariner. Have eliminated this chunk of code since the root cause of the failure of mig-partition.service is related to timeout.